driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -860,7 +1202,59 @@ SystemRows fetchSystemRows(
         VersionNumber cassandraVersion)
         throws ConnectionException, BusyConnectionException, ExecutionException,
             InterruptedException {
+      if (targetType == null
+          && cluster.getConfiguration().getQueryOptions().isSchemaQueriesPaged()) {
+        Map<String, KeyspaceMetadata> keyspaces =
+            buildSchema(cluster, connection, cassandraVersion);
+        Metadata metadata;
+        try {
+          metadata = cluster.getMetadata();
+        } catch (IllegalStateException e) {
+          logger.warn("Unable to refresh metadata, cluster has been closed");
+          return;
+        }
+        metadata.lock.lock();
+        try {
+          updateKeyspaces(metadata, metadata.keyspaces, keyspaces, null);
+          // If we rebuild all from scratch or have an updated keyspace, rebuild the token map
+          // since some replication on some keyspace may have changed
+          metadata.rebuildTokenMap();
+        } catch (RuntimeException e) {
```

Comment:
Why only catch `RuntimeException` here not also `ConnectionException, BusyConnectionException, ExecutionException, InterruptedException`? Why not include

```
 Map<String, KeyspaceMetadata> keyspaces =
            buildSchema(cluster, connection, cassandraVersion);
```
in this `try {}` block (as `buildSchema` can throw)?